src/base/settingvalue.h

```diff
@@ -68,11 +69,44 @@ class CachedSettingValue
     CachedSettingValue<T> &operator=(const T &newValue)
     {
         m_value = newValue;
-        SettingsStorage::instance()->storeValue(m_keyName, m_value);
+        storeValue(m_value);
         return *this;
     }
 
 private:
+    // regular load/save pair
+    template <typename U, typename std::enable_if<!std::is_enum<U>::value, int>::type = 0>
+    U loadValue(const U &defaultValue)
+    {
+        return SettingsStorage::instance()->loadValue(m_keyName, defaultValue).template value<T>();
+    }
+
+    template <typename U, typename std::enable_if<!std::is_enum<U>::value, int>::type = 0>
+    void storeValue(const U &value)
+    {
+        SettingsStorage::instance()->storeValue(m_keyName, value);
+    }
+
+    // load/save pair for enum
+    // saves literal value of the enum constant, obtained from QMetaEnum
+    template <typename U, typename std::enable_if<std::is_enum<U>::value, int>::type = 0>
+    U loadValue(const U &defaultValue)
+    {
+        bool ok = false;
+        const QString sval = SettingsStorage::instance()->loadValue(m_keyName, QString()).toString();
+        const int val = QMetaEnum::fromType<U>().keyToValue(sval
+            .toLatin1().constData(), &ok);
+        const U res = static_cast<U>(val);
+        return ok ? res : defaultValue;
+    }
+
+    template <typename U, typename std::enable_if<std::is_enum<U>::value, int>::type = 0>
+    void storeValue(const U &value)
+    {
+        SettingsStorage::instance()->storeValue(m_keyName,
```

Comment:
It might make sense to use `std::underlying_type` here and assert that enum underlying type is `int`.